github/github.go

```diff
@@ -235,6 +235,23 @@ func NewClient(httpClient *http.Client) *Client {
 	return c
 }
 
+// NewEnterpriseClient returns a new GitHub API client (same as NewClient),
+// except it allows injection of custom base and upload urls. These urls might
+// be the same, but for flexibility both can be set separately.
+func NewEnterpriseClient(httpClient *http.Client, baseURL string, uploadURL string) *Client {
+	if !strings.HasSuffix(baseURL, "/") {
+		baseURL = baseURL + "/"
+	}
+	if !strings.HasSuffix(uploadURL, "/") {
+		uploadURL = uploadURL + "/"
+	}
```

Comment:
How about modifying the `Path` field of the URL after parsing it?

Also, why not use `+=` operator?

```Go
if !strings.HasSuffix(client.BaseURL.Path, "/") {
	client.BaseURL.Path += "/"
}
if !strings.HasSuffix(client.UploadURL.Path, "/") {
	client.UploadURL.Path += "/"
}
```

Reply:
Totally agree about the +=, is much better (my golang inexperience coming through :D).

No particular reason I did it this way around. However, given the discussion about returning the error, perhaps it makes more sense to parse as early as possible and edit the Path only after it is known to be valid.